clients/drcachesim/tracer/raw2trace.cpp

```diff
@@ -510,6 +528,13 @@ raw2trace_t::merge_and_process_thread_files()
                 FATAL_ERROR("Failed to write to output file");
         }
     } while (file_count > 0);
+    // If we used threading pool during profiling, the trace for a thread may be
+    // seen after the thread exit entry, so we can only check whether the unique
+    // tid and tid exit matches.
+    VPRINT(1, "%u thread(s) are seen\n", (uint)tid_set.size());
+    if (tid_set != tid_exit_set) {
+        WARN("Profile might be corrupted.");
```

Comment:
A more specific message, please!  Again the prior code was IMHO better: "Trace for thread %d is truncated".